File: ITCTCBackend/src/main/java/com/synergiz/itctc/service/CamMeasurementService.java
package com.synergiz.itctc.service;


import com.synergiz.itctc.dto.request.CamMeasurementRequest;
import com.synergiz.itctc.dto.response.CamMeasurementResponse;

import java.util.List;

public interface CamMeasurementService {

   
    Long saveCamMeasurement(CamMeasurementRequest request);

  
    CamMeasurementResponse getCamMeasurement(Long camMeasurementHeaderId);


    List<CamMeasurementResponse> getAllCamMeasurements();


    Long updateCamMeasurement(
            Long camMeasurementHeaderId,
            CamMeasurementRequest request);


    void deleteCamMeasurement(
            Long camMeasurementHeaderId,
            String updatedBy);

}

File: ITCTCFrontend/src/services/camMeasurementService.js
import api from "./api";


export const getAllCamMeasurements = async () => {
    const response = await api.get("/cam-measurements");
    return response.data;
};


export const getCamMeasurementById = async (id) => {
    const response = await api.get(`/cam-measurements/${id}`);
    return response.data;
};


export const saveCamMeasurement = async (data) => {
    const response = await api.post("/cam-measurements", data);
    return response.data;
};


export const updateCamMeasurement = async (id, data) => {
    const response = await api.put(`/cam-measurements/${id}`, data);
    return response.data;
};


export const deleteCamMeasurement = async (id,updatedBy) => {
    const response = await api.delete(`/cam-measurements/${id}?updatedBy=${updatedBy}`);
    return response.data;
};